### PR TITLE
refactor!: Rename Binder.Binding.value() to valueSignal() and return signal

### PR DESCRIPTION
**Pull request by mshabarov**
Reworks Binding.value() to return a ValueSignal<TARGET> instead of the raw value, and renames the method to valueSignal() to reflect this change.

The valueSignal is lazily initialized and updated whenever:
- The field value changes through user interaction
- The field value is set from a bean (via setBean/readBean)

Simplifies the implementation by removing internalValidationTriggerSignal and reusing valueSignal for tracking purposes. When validators call valueSignal().get(), they automatically track the signal, and updates to the signal trigger re-validation without needing a separate boolean toggle signal.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by mshabarov** [comment]
Validated with [cross-field password confirmation](https://github.com/vaadin/signals-cases/blob/a4c7dfd0f926549f7057613de2a18461567e5f29/src/main/java/com/example/usecase09/UseCase09View.java#L98)

File: flow-data/src/main/java/com/vaadin/flow/data/binder/Binder.java [hunk 1]
```diff
@@ -1513,7 +1512,7 @@ protected static class BindingImpl<BEAN, FIELDVALUE, TARGET>
 
         private transient Registration signalRegistration;
 
-        private transient ValueSignal<Boolean> internalValidationTriggerSignal;
+        private transient ValueSignal<TARGET> valueSignal;
```
Comment (tltv) [inline flow-data/src/main/java/com/vaadin/flow/data/binder/Binder.java]: Should it be called `triggerValueSignal` just to make a difference with bindValue's signal? Having "Trigger" in the name would better point out its purpose in the binding and make it easier to understand its not same signal as given to the field's `bindValue`.

Comment (tltv) [reply flow-data/src/main/java/com/vaadin/flow/data/binder/Binder.java]: ...and it can't be the same `ValueSignal` instance anyway since one can call `bindValue` after Binder is called with `readBean`.

Comment (mshabarov) [reply flow-data/src/main/java/com/vaadin/flow/data/binder/Binder.java]: I'd call it `bindingValueSignal`, but it doesn't help much to distinguish the semantics from `bindValue`. "Trigger" doesn't look correct, because it's now not just an internal trigger signal, but a signal that can be used in the project, e.g. to subscribe to the field updates.

Comment (tltv) [reply flow-data/src/main/java/com/vaadin/flow/data/binder/Binder.java]: Its primary use case is for cross-field validation to work as a trigger for re-validation. I'm not aware of other use-cases. If there is other  use cases, then Javadoc should be updated too. I'm fine with `bindingValueSignal` name too.

Comment (mshabarov) [reply flow-data/src/main/java/com/vaadin/flow/data/binder/Binder.java]: > ...and it can't be the same `ValueSignal` instance anyway since one can call `bindValue` after Binder is called with `readBean`.

Don't get it. How `bindValue` signal is related to binding's signal? 
If I say
```
ValueSignal<String> passwordSignal = new ValueSignal<>("");
passwordField.bindValue(passwordSignal, passwordSignal::set);
binder.forField(confirmField)
        .withValidator(text -> text.equals(passwordSignal.get()),
               "Both fields must match")
        .bind("confirmPassword");
```
then it will be a different signal.

Comment (mshabarov) [reply flow-data/src/main/java/com/vaadin/flow/data/binder/Binder.java]: > Its primary use case is for cross-field validation to work as a trigger for re-validation. I'm not aware of other use-cases. If there is other use cases, then Javadoc should be updated too. I'm fine with `bindingValueSignal` name too.

Ok, let's rename to `bindingValueSignal`.

Comment (tltv) [reply flow-data/src/main/java/com/vaadin/flow/data/binder/Binder.java]: > Don't get it. How bindValue signal is related to binding's signal?

It's not. But when I reviewed the code I though that it is related and therefore I asked the name change suggestion to make it a bit clearer that there's no connection.

**Review by tltv — COMMENTED** [review]
Consider adding following new test in `BinderSignalTest` class. It was missing originally and what it does, it counts validator calls and asserts them. It checks especially how `binding.valueSignal().get()` call behave. Currently it triggers re-validation on each `Signal#get()` dependency even with same `ValueSignal` instance. This is quite an edge case so it's good to have a test that verifies how it works currently:

```java
    @Test
    public void bindingValue_multipleValueSignalCalls_revalidateTriggeredForEach() {
        item.setFirstName("Alice");
        item.setLastName("Smith");

        AtomicInteger validatorCalls = new AtomicInteger(0);
        
        UI.getCurrent().add(firstNameField, lastNameField);
        var lastNameBinding = binder.forField(lastNameField).bind("lastName");
        binder.forField(firstNameField)
                .withValidator((String value) -> {
                    validatorCalls.incrementAndGet();
                    return !value.isEmpty()
                            // multiple calls to same valueSignal()
                            && !lastNameBinding.valueSignal().get().isEmpty()
                            && !lastNameBinding.valueSignal().get().isEmpty();
                },
                        "First and last name are required")
                .bind("firstName");
        assertEquals(1, validatorCalls.get());

        binder.setBean(item);

        assertEquals(2, validatorCalls.get());

        assertTrue(binder.isValid());

        assertEquals(3, validatorCalls.get());

        lastNameField.setValue("");
        // value change runs revalidation for both Signal.get() calls in the validator
        assertEquals(5, validatorCalls.get());
    }
```

**Review by tltv — DISMISSED** [review]

**Comment by mshabarov** [comment]
> Consider adding following new test in `BinderSignalTest` class. It was missing originally and what it does, it counts validator calls and asserts them. It checks especially how `binding.valueSignal().get()` call behave. Currently it triggers re-validation on each `Signal#get()` dependency even with same `ValueSignal` instance. This is quite an edge case so it's good to have a test that verifies how it works currently: (Edit: removed one assert that slipped in wrong place)

Thanks! Added.
